src/content/app/entity-viewer/variant-view/transcript-consequences/transcript-consequence-details/TranscriptConsequenceDetails.tsx

```diff
@@ -0,0 +1,146 @@
+import styles from './TranscriptConsequenceDetails.module.css';
+
+type Props = {
+  genomeId: string;
+  transcriptId: string;
+  gene: TranscriptConsequencesData['geneData'];
+  variant: TranscriptConsequencesData['variant'];
+  allele: NonNullable<TranscriptConsequencesData['allele']>;
+};
+
+const TranscriptConsequenceDetails = (props: Props) => {
+  const { gene, variant, allele } = props;
+
+  const { currentData: transcriptDetailsData, isLoading } =
+    useTranscriptDetails(props);
+
+  const transcript = transcriptDetailsData?.transcriptData;
+  const genomicRegionData = transcriptDetailsData?.genomicRegionData;
+
+  if (isLoading) {
+    return (
+      <div className={commonStyles.row}>
+        <div className={commonStyles.middle}>
+          <CircleLoader />
+        </div>
+      </div>
+    );
+  } else if (!transcript || !genomicRegionData) {
+    return null;
+  }
+
+  const strand = gene.slice.strand.code;
+
+  return (
+    <>
+      <div className={commonStyles.row}>
+        <div className={commonStyles.left}>Genomic</div>
+        <div className={commonStyles.middle}>
+          <VariantPositionInTranscript
+            distanceToTranscriptStart={
+              genomicRegionData.variantToTranscriptStartDistance
+            }
+            referenceAlleleSequence={genomicRegionData.referenceAlleleSequence}
+            transcriptLength={transcript.slice.location.length}
+          />
+        </div>
+      </div>
+
+      <div className={commonStyles.row}>
+        <div className={commonStyles.middle}>
+          <TranscriptVariantGenomicSequence
+            {...genomicRegionData}
+            variantSequence={genomicRegionData.referenceAlleleSequence}
+            variantType={variant.allele_type.value}
+            alleleType={allele.allele_type.value}
+          />
+        </div>
+        <div className={commonStyles.right}>
+          <span className={styles.smallLight}>
+            {strand === 'forward' ? 'forward strand' : 'reverse complement'}
+          </span>
+        </div>
+      </div>
+
+      <div className={commonStyles.row}>
+        <div
+          className={classNames(
+            commonStyles.left,
+            styles.transcriptDiagramLabel
+          )}
+        >
+          Transcript position in gene
+        </div>
+        <div className={commonStyles.middle}>
+          <TranscriptVariantDiagram
+            gene={gene}
+            transcript={transcript}
+            variant={variant}
+          />
+        </div>
+      </div>
+    </>
+  );
+};
+
+const VariantPositionInTranscript = ({
+  distanceToTranscriptStart,
+  referenceAlleleSequence,
+  transcriptLength
+}: {
+  distanceToTranscriptStart: number;
+  referenceAlleleSequence: string;
+  transcriptLength: number;
+}) => {
+  const variantLength = referenceAlleleSequence.length;
+  const variantStartInTranscript = distanceToTranscriptStart + 1;
+  const variantEndInTranscript = variantStartInTranscript + variantLength;
```

Comment:
Shouldn't this be subtracted by 1? For example, if variant start is 207 and variant length is 12 then shouldn't the end be 218 (207 + 12 - 1)?

Reply:
Yes, you are right. I was probably thinking of insertions when I was writing this line. Now, for insertions, the length of the reference allele, as we represent it on the website, is 0. So an extra guard needs to be put into place to make sure variant end is never lower than variant start:

```
  const variantEndInTranscript = Math.max(
    variantStartInTranscript + variantLength - 1,
    variantStartInTranscript
  );
```